From ec993a6e61e582efdc6622724115ae5360a391d3 Mon Sep 17 00:00:00 2001
From: WangFengTu <wangfengtu@huawei.com>
Date: Wed, 13 Nov 2019 11:27:55 +0800
Subject: [PATCH 32/37] Fix docker pull failed caused by unexpected key

Temporary created key should not be used to match cert.
Ignore the key by adding a prefix as a filter.

Signed-off-by: WangFengTu <wangfengtu@huawei.com>
---
 .../image/pkg/tlsclientconfig/tlsclientconfig.go       | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/vendor/github.com/containers/image/pkg/tlsclientconfig/tlsclientconfig.go b/vendor/github.com/containers/image/pkg/tlsclientconfig/tlsclientconfig.go
index b564e23..89ca1ec 100644
--- a/vendor/github.com/containers/image/pkg/tlsclientconfig/tlsclientconfig.go
+++ b/vendor/github.com/containers/image/pkg/tlsclientconfig/tlsclientconfig.go
@@ -1,7 +1,9 @@
 package tlsclientconfig
 
 import (
+	"crypto/rand"
 	"crypto/tls"
+	"fmt"
 	"io/ioutil"
 	"net"
 	"net/http"
@@ -9,9 +11,6 @@ import (
 	"path/filepath"
 	"strings"
 	"time"
-	"fmt"
-	"crypto/rand"
-
 
 	"github.com/docker/go-connections/sockets"
 	"github.com/docker/go-connections/tlsconfig"
@@ -28,6 +27,7 @@ var (
 	ErrAlreadyExists = errors.New("Image already exists")
 	psmPath          = "/etc/isulad/psm"
 	IsDecrypted      = false
+	tmpKeyPrefix     = ".tmp_origin"
 )
 
 func decodeKeyFile(key, originKey string, directory string) error {
@@ -127,7 +127,7 @@ func SetupCertificates(dir string, tlsc *tls.Config, IsDecrypted bool) error {
 			keyOriginPath := keyPath
 
 			if !IsDecrypted {
-				keyOriginPath = filepath.Join(dir, "origin"+randStr(10)+keyName)
+				keyOriginPath = filepath.Join(dir, tmpKeyPrefix+randStr(10)+keyName)
 				defer os.Remove(keyOriginPath)
 				if err := decodeKeyFile(keyPath, keyOriginPath, dir); err != nil {
 					return err
@@ -140,7 +140,7 @@ func SetupCertificates(dir string, tlsc *tls.Config, IsDecrypted bool) error {
 			}
 			tlsc.Certificates = append(tlsc.Certificates, cert)
 		}
-		if strings.HasSuffix(f.Name(), ".key") {
+		if strings.HasSuffix(f.Name(), ".key") && !strings.HasPrefix(f.Name(), tmpKeyPrefix) {
 			keyName := f.Name()
 			certName := keyName[:len(keyName)-4] + ".cert"
 			logrus.Debugf(" key: %s", fullPath)
-- 
2.19.1

